web3pi_proxy/core/rpc/node/endpoint_pool/pool_manager.py

```diff
@@ -92,14 +90,114 @@ def check_connections(self, pools: List[EndpointConnectionPool]):
                     daemon=True,
                 ).start()
 
-            if self.__is_outdated(pool):
-                self.__logger.warning(f"Endpoint `{pool.endpoint.name}`: falling behind")
-
             self.__logger.debug(f"Endpoint `{pool.endpoint.name}`: okay.")
 
 
+class HttpResponseParserListenerSyncController:
+    body: bytes = b''
+    block_number: int = 0
+    block_timestamp: int = 0
+    __logger = get_logger("SyncController")
+
+    def __init__(self, __pool_name: str):
+        super().__init__()
+        self.__pool_name = __pool_name
+
+    def on_body(self, body: bytes):
+        self.body = self.body + body
+
+    def on_message_complete(self):
+        try:
+            block_data = json.loads(self.body)
+            result = block_data.get('result')
+            if not result:
+                error = block_data.get('error')
+                if error:
+                    self.__logger.error(f"{self.__pool_name}: Sync test failed: {error}")
+                else:
+                    self.__logger.error(f"{self.__pool_name}: Sync test failed: reason unknown")
+                return
+            self.block_number = int(result['number'], 16)
+            self.block_timestamp = int(result['timestamp'], 16)
+        except Exception as error:
+            self.__logger.error("%s: %s", error.__class__, error)
+            self.__logger.error(f"{self.__pool_name}: Failed to failed to parse sync test response")
+
+
+# TODO: Tune parameters
+class SyncController:
+    __SUSPENSION_TIMEOUT_SECONDS = 60
+    __MAX_DELAY_SECONDS = 60
+
+    __logger = get_logger("SyncController")
+
+    def __test_pool(self, pool: EndpointConnectionPool) -> bool | None:
+        endpoint_connection_handler: EndpointConnectionHandler | None = None
+        try:
+            endpoint_connection_handler = pool.get(out_of_sync=True)
+            req = RPCRequest()
+            req.headers = b'User-Agent: curl/8.0.1\r\nAccept: */*\r\nContent-Type: application/json\r\nContent-Length: 82\r\n'
```

Comment:
plus, please don't use `curl` as a user-agent, when performing your own, custom query ;) I believe it's legal to omit this header altogether... but if we do want to put it in, let's add e.g. `web3pi-proxy/[version]` or something similar...